sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Diagnostics/AzureMonitorExporterEventSource.New.cs

```diff
@@ -153,12 +141,75 @@ public void FailedToExtractActivityEvent(string activitySourceName, string activ
         public void FailedToExtractActivityEvent(string activitySourceName, string activityDisplayName, string exceptionMessage) => WriteEvent(17, activitySourceName, activityDisplayName, exceptionMessage);
 
         [Event(18, Message = "Maximum count of {0} Activity Links reached. Excess Links are dropped. ActivitySource: {1}. Activity: {2}.", Level = EventLevel.Warning)]
-        public void ActivityLinksIgnored(int maxLinksAllowed, string activitySourceName, string activityDisplayName)
+        public void ActivityLinksIgnored(int maxLinksAllowed, string activitySourceName, string activityDisplayName) => WriteEvent(18, maxLinksAllowed, activitySourceName, activityDisplayName);
+
+        [Event(19, Message = "Failed to parse redirect headers. Not user actionable.", Level = EventLevel.Informational)]
+        public void RedirectHeaderParseFailed() => WriteEvent(19);
+
+        [Event(20, Message = "Failed to parse redirect cache, using default. Not user actionable.", Level = EventLevel.Informational)]
+        public void ParseRedirectCacheFailed() => WriteEvent(20);
+
+        [NonEvent]
+        public void ErrorCreatingStorageFolder(string path, Exception ex)
+        {
+            if (IsEnabled(EventLevel.Error))
+            {
+                ErrorCreatingStorageFolder(path, ex.FlattenException().ToInvariantString());
+            }
+        }
+
+        [Event(21, Message = "Failed to create a storage directory at path '{0}' due to an exception. If a storage directory cannot be created, telemetry may be lost. {1}", Level = EventLevel.Error)]
+        public void ErrorCreatingStorageFolder(string path, string exceptionMessage) => WriteEvent(21, path, exceptionMessage);
+
+        [NonEvent]
+        public void ErrorInitializingRoleInstanceToHostName(Exception ex)
+        {
+            if (IsEnabled(EventLevel.Error))
+            {
+                ErrorInitializingRoleInstanceToHostName(ex.FlattenException().ToInvariantString());
+            }
+        }
+
+        [Event(22, Message = "Failed to initialize Role Instance due to an exception. This data will be missing from telemetry. {0}", Level = EventLevel.Error)]
+        public void ErrorInitializingRoleInstanceToHostName(string exceptionMessage) => WriteEvent(22, exceptionMessage);
+
+        [NonEvent]
+        public void ErrorInitializingPartOfSdkVersion(string typeName, Exception ex)
+        {
+            if (IsEnabled(EventLevel.Error))
+            {
+                ErrorInitializingPartOfSdkVersion(typeName, ex.FlattenException().ToInvariantString());
+            }
+        }
+
+        [Event(23, Message = "Failed to get Type version while initialize SDK version due to an exception. Not user actionable. Type: {0}. {1}", Level = EventLevel.Error)]
```

Comment:
If it does not impact, let mark the event level as warning. Same for `SdkVersionCreateFailed`